core/templates/services/image-sanitizer.service.ts

```diff
@@ -174,7 +174,24 @@ export class SvgSanitizerService {
     }
     return null;
   }
+
+  isValidBase64Png(base64ImageData: string): boolean {
+    const DATA_URL_PATTERN = /^data:image\/png;base64,[a-z0-9+\/%]+=*$/i;
```

Comment:
The regex for base64 is not valid. `%` is not a valid base64 character. If the image data is containing `%`, it would imply that the image is URI-encoded. For example, `=` becomes `%3D`, `+` becomes `%2B` and `/` becomes `%2F`. If this is the case with the images, then decode the image using the inbuilt Javascript function `decodeURIComponent` and the image will pass the angular sanity checks.

Reply:
Thanks for the explanation Srijan. Made the required changes.